marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserArtifactsCommand.java

```diff
@@ -188,44 +201,45 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
                     }
                 });
             }
-
-            // let's do step-definitions
-            if (stepDefPath.toFile().exists()) {
-                Files.walkFileTree(stepDefPath, new SimpleFileVisitor<Path>() {
-                    @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        executeWalk(
-                            dir,
-                            stepDefModulesFinder,
-                            propertiesModuleManager,
-                            stepResourceToURI,
-                            buildMetadata(hubConfig.getStepDefinitionPermissions(), "http://marklogic.com/data-hub/step-definition"),
-                            stagingStepDefDocumentWriteSet,
-                            finalStepDefDocumentWriteSet
-                        );
-                        return FileVisitResult.CONTINUE;
-                    }
-                });
-            }
-
-
-            // let's do flows
-            if (flowPath.toFile().exists()) {
-                Files.walkFileTree(flowPath, new SimpleFileVisitor<Path>() {
-                    @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        executeWalk(
-                            dir,
-                            flowDefModulesFinder,
-                            propertiesModuleManager,
-                            flowResourceToURI,
-                            buildMetadata(hubConfig.getFlowPermissions(), "http://marklogic.com/data-hub/flow"),
-                            stagingFlowDocumentWriteSet,
-                            finalFlowDocumentWriteSet
-                        );
-                        return FileVisitResult.CONTINUE;
-                    }
-                });
+            ArtifactManager artifactManager = new ArtifactManagerImpl(hubConfig);
+            List<String> roles = new LinkedList<>();
+            RolesService.on(stagingClient).getRoles().elements().forEachRemaining(
+                (roleNode) -> roles.add(roleNode.asText())
+            );
+            ArtifactService artifactService = ArtifactService.on(stagingClient);
+            ObjectMapper mapper = new ObjectMapper();
+            // New way to deploy Artifacts via Data Services
+            for (ArtifactTypeInfo typeInfo: artifactManager.getArtifactTypeInfoList()) {
+                final String artifactType = typeInfo.getType();
+                final Path artifactPath = hubConfig.getHubProject().getProjectDir().toAbsolutePath().resolve(typeInfo.getDirectory().replaceAll("^/","")).toAbsolutePath();
+                logger.info("Deploying artifacts of type '" + artifactType + "' located at '"+ artifactPath.toString() +"'" );
+                if (roles.contains(typeInfo.getUpdateRole()) && artifactPath.toFile().exists()) {
```

Comment:
put the following right after for loop
`
if (roles.contains(typeInfo.getUpdateRole()) {
     continue;
 }
`

Reply:
I was thinking - first check to see if the artifactPath doesn't exist, and if not, log an info message of "Artifact directory (directory) does not exist, so will not load any artifacts from it". Then check to see if the user has the update role, and if so, load the artifacts; if not, log an info message stating "User is not permitted to update artifacts of type (type), so will not load any artifacts of that type".

Reply:
Changed to use continue calls and added more specific logging as to why an artifact type may not be loaded.